src/commands/org/create/sandbox.ts

```diff
@@ -262,17 +291,13 @@ export default class CreateSandbox extends SandboxCommandBase<SandboxCommandResp
         this.flags.wait.seconds,
       ]);
     }
-  }
 
-  private async getSourceId(): Promise<string | undefined> {
-    if (!this.flags.clone) {
-      return undefined;
-    }
-    try {
-      const sourceOrg = await this.flags['target-org'].querySandboxProcessBySandboxName(this.flags.clone);
-      return sourceOrg.SandboxInfoId;
-    } catch (err) {
-      throw messages.createError('error.noCloneSource', [this.flags.clone], [], err as Error);
+    const defFileContent = this.flags['definition-file'] ? readSandboxDefFile(this.flags['definition-file']) : {};
```

Comment:
nit: you could do an early check if `--definition-file` isn't passed and skip the validation (`--source-sandbox-name`  defines has `source-id` as an exclusive flag so oclif will handle user don't pass both)